aws/resource_aws_cognito_identity_pool.go

```diff
@@ -164,10 +170,11 @@ func resourceAwsCognitoIdentityPoolRead(d *schema.ResourceData, meta interface{}
 		AccountID: meta.(*AWSClient).accountid,
 		Resource:  fmt.Sprintf("identitypool/%s", d.Id()),
 	}
-	d.Set("arn", arn.String())
-	d.Set("identity_pool_name", ip.IdentityPoolName)
-	d.Set("allow_unauthenticated_identities", ip.AllowUnauthenticatedIdentities)
-	d.Set("developer_provider_name", ip.DeveloperProviderName)
+	_ = d.Set("arn", arn.String())
+	_ = d.Set("identity_pool_name", ip.IdentityPoolName)
+	_ = d.Set("allow_unauthenticated_identities", ip.AllowUnauthenticatedIdentities)
+	_ = d.Set("developer_provider_name", ip.DeveloperProviderName)
+	_ = d.Set("tags", tagsToMapGeneric(ip.IdentityPoolTags))
```

Comment:
This is the only one here where we would recommend the error check. When using `d.Set()` with aggregate types (TypeList, TypeSet, TypeMap), [we should perform error checking to prevent issues where the code is not properly able to set the Terraform state](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#error-checking-aggregate-types). e.g.

```suggestion
	if err := d.Set("tags", tagsToMapGeneric(ip.IdentityPoolTags)); err != nil {
		return fmt.Errorf("error setting tags: %s", err)
	}
```

Reply:
fixed in 4ed6457ff